### DISCUSSION TITLE
RuntimeError: bad allocation

### DISCUSSION THREAD

**Discussion by solo707** [Chia-Network/chia-blockchain#1847] [Q&A] [answered]
Starting phase 1/4: Forward Propagation into tmp files... Tue Apr 13 21:23:51 2021
Computing table 1
F1 complete, time: 576.03 seconds. CPU (181.01%) Tue Apr 13 21:33:27 2021
Computing table 2
Caught plotting error: bad allocation
[14772] Failed to execute script chia
Traceback (most recent call last):
  File "chia\cmds\chia.py", line 80, in <module>
  File "chia\cmds\chia.py", line 76, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "chia\cmds\plots.py", line 134, in create_cmd
  File "chia\plotting\create_plots.py", line 176, in create_plots
RuntimeError: bad allocation

my plots stop at 1%，with these words

**Comment by hoffmang9** [comment, marked as answer]
You have a problem with your RAM. It could be related to your swap file or it can be actual hardware issues with your RAM.

**Reply by solo707** [reply]
I find the problem，thank you！

**Reply by hag400** [reply]
Hi solo707, how did you solve your problem?  I'm getting the same error particularly on one Samsung EVO SSD.

**Reply by ericiggy** [reply]
SAME PROBLEM!

**Reply by CraftInsights** [reply]
I tried to queue up another plot and I got this error. Please let me know how you resolved the issue.

**Reply by hoffmang9** [reply]
https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#my-plotting-attempt-ended-with-runtimeerror-bad-allocation

**Reply by Andrey5045** [reply]
Same problem at 1% of k=35, looks like it is OK @ k=34.

**Comment by glassclow3** [comment]
Also I need help here, I dont know how to fix this RuntimeError: Unknown exception.
I'm new to this chia mining app and doesn't know how to work this thing to get chia T_T


Traceback (most recent call last):
  File "chia\cmds\chia.py", line 80, in <module>
  File "chia\cmds\chia.py", line 76, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "chia\cmds\plots.py", line 134, in create_cmd
  File "chia\plotting\create_plots.py", line 176, in create_plots
RuntimeError: Unknown exception

**Reply by hag400** [reply]
How many CPU cores and how much memory do you have?  Also what settings are you using when you plot?

**Reply by ericiggy** [reply]
SAME!

**Comment by Orozuz** [comment]
I'm having a problem too, says "caught ploting error: Could not open" then the location of the file and then "Permission denied" and finally "RuntimeError: Unknown exception"
I have no idea what to do

**Comment by VaverDariush** [comment]
For me it was a swap issue. I changed windows settings to have constant swap equal to roughly double ram on my SSD drive and the error stopped appearing.

**Reply by zjjne** [reply]
Thanks. I solved by setting the Windows Virtual Memory value.

**Reply by kindofausername** [reply]
Yes, setting fixed swap sizes for all plotting ssds did the trick.

**Comment by burakco** [comment]
I have that same problem. The 5th plot gives this error. Do I have to start the process over or wait for the other plots to finish?

**Reply by AspiPL** [reply]
Same problem on my machine - 4 plots are being plotted OK, anything more crashes. Checked RAM with memtest - 4 passes ok. 64 gigs. CPU not overclocked (Ryzen 9 3950x).

**Reply by tarzanking** [reply]
@burakco how was your result 
u wait for the other plots to finish or reprocess it again?

**Comment by snafusam** [comment]
I am getting the same problem, this is actually my first plot using a ryzen 9 3900x (12 cores allocated) and 16 gb 3200 g.skill ripjaws (6750mb allocated). Running 4 plots in parallel. Is there a way to stop this process and start over or should I let it finish?
![Annotation 2021-04-22 050502](https://user-images.githubusercontent.com/69125345/115711765-cab7e200-a328-11eb-873f-e459d15eae6c.png)

**Reply by tarzanking** [reply]
i have same trouble with u 
and now one of the port is having issue as well ...
4 port become one port running

**Comment by kelatoz** [comment]
reboot ？

**Comment by tal4ek** [comment]
And i have this problem, but i don't resolved ((

**Reply by hoffmang9** [reply]
https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#my-plotting-attempt-ended-with-runtimeerror-bad-allocation

**Comment by parko86** [comment]
For me, it's the CPU overclock too aggressive. Hope this can help some of you with this problem.

**Comment by PovAndy** [comment]
I found a solution to the problem on this site! [How To Fix Chia ‘RunTime Error: Bad Allocation’ On Windows] (https://wccftech.com/how-to-fix-chia-runtime-error-bad-allocation-on-windows/)
Try it, then write, did it help you?

**Comment by atus42** [comment]
Well, I got this error on windows 10 several times at the start of phase 3! (compressing tables 1 and 2) 10 hours of waste of time and ssd writes.
Only 2 plots parralel, with 16G total ram and param -b 3400, so there should be enough ram. Changed thw swap file to fixed 5120MB.
On a poor linux PC with the same 16G ram, I never got this error in parralel plotting.

**Reply by PovAndy** [reply]
On Windows, the system wastes more memory.
With 16GB of memory, 3 parallel plotting processes must be run on 128 buckets and 3400 memory.
3.400GiB = 3.650Gb
For 3 parallel plotting processes -> 3.650Gb * 3 = 10.950Gb 

What is your SSD size?
Probably not enough space for 3 threads!

**Reply by atus42** [reply]
As I wrote, there is only two parallel plotting processes. For the very reason to be on the safe side. And the error happens at phase three! not at phase one, computing table two.

My nvme is 500Gb, so I have to delay the starts to avoid running out of space. That is another problem, and another error shown on the command line. This bad allocation error I got is alwas come at the start of phase 3. If I check the free space, there have always more than a GB.

**Reply by PovAndy** [reply]
This is the answer to your question!
A minimum of 256Gb * 3 = 768GB SSD is required for 3 parallel build processes !!!

When setting up drawing, set the 2nd temporary directory on the HDD, it will help you!

**Reply by atus42** [reply]
Space consumption of plotting is like a wave. It stars from zero, goes up to 2.5*plot size, and then slowly falls down to plot size.

I saw here and on another threads that the amount of virtual memory coud help, so I changed the the virtual memory settings a little bit. There was a 2854MB swap on a regular HDD D: and ther was no swap on drive C: (nvme). I changed the swap size to 5120MB on D: and added another 5120MB on C:. I dropped the -b 3400 parameter from the command line, and start **two** parallel plots again **with a 30 minute time shift**. The firs started at 10:30, the second at 11:00. They both succesfully survived the begining of phase 3. 
As of now, 6.5 hour later the firs one is compressing table 4 and five, the other compressing 3 and 4. Both have a temp dir on the before mentioned 500G nvme card. The least free space the disk had was at the begining of phase 3 (plot 1). There was around 20G free space at that time. Now, it is went up to 122G.

**Reply by PovAndy** [reply]
500 Gb is not enough for 2 processes in parallel! (A 500Gb SSD has a real size of about 480-490Gb!)
For example, on a 1Tb SSD, you can run only 3 threads in parallel!
When setting up drawing, set the 2nd temporary directory on the HDD, it will help you!
How much RAM do you have?

**Reply by atus42** [reply]
Well... both plotting process finished successfully. I think the default memory usage and maybe the increased amount of swap solved this.

**Reply by PovAndy** [reply]
If you don't have enough RAM, this could be the reason!

**Reply by Sent78** [reply]
Hello. And if I have a 1 tb ssd, and the swap file is on the C drive of 8192 mb, and 16 GB of RAM, the number of seeding is 3 in parallel. And the error comes out the same. Where to dig with Windows settings?

**Comment by AudioCrafter** [comment]
I have this error while plotting and mining with my GPU at the same time. If i close my GPU miner i dont have this error.
